### PR TITLE
Baseband fw tp yoga 12

### PR DESCRIPTION
**Pull request by senier**
Port of the baseband demo to the Lenovo ThinkPad Yoga 12. This now works with touch input. Please test on Yoga-900.

### CODE REVIEW

**Comment by jklmnn** [comment]
I couldn't test the run files yet since I'm unable to build the rump tool chain.
Running `tool/tool_chain_rump build ` fails with 
```
In file included from /.../contrib/rumprun/rumpsrc/lib/librumpuser/rumpuser_port.h:61:0,
                 from /.../contrib/rumprun/rumpsrc/lib/librumpuser/rumpuser_dl.c:40:
/usr/include/features.h:148:3: error: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Werror=cpp]
 # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
   ^~~~~~~
```
This warning comes from libc6-dev (2.24-11+deb9u1) so I currently can't change anything about this. How did you build the rump tool chain?

File: run/baseband_demo.inc [hunk 1]
```diff
@@ -83,7 +89,12 @@ append config {
 		</provides>
 		<config ehci="yes" uhci="yes" ohci="yes" xhci="yes">
 			<storage/>
-			<hid/>
+			<hid>}
```
[the remark this answers is not in the document]
Comment (senier) [reply run/baseband_demo.inc]: Agreed. This still wouldn't make run scripts easy to read and use. I want to replace them with our own XML-only files (cf. https://github.com/Componolit/componolit/issues/70).

**Comment by senier** [comment]
I had the same issue - this seems to be broken right now. Eventually I realized that that build script should not be necessary. The `dde_rump` port does this on its own. Does `make run/rump_ext2` work for you? (You need to disable `ccache`!)

**Comment by jklmnn** [comment]
`run/rump_ext2` fails to build with
```
rump.lib.so: undefined reference to `abort'
rump.lib.so: undefined reference to `malloc'
rump.lib.so: undefined reference to `free'
rump.lib.so: undefined reference to `strlen'
collect2: error: ld returned 1 exit status
```

The Yoga build fails with the same error now. Interestingly I can't reproduce the first error even after running `make cleanall` and deleting all build artifacts.

**Comment by senier** [comment]
My fault. Please retry with my latest commit on the componolit-stable branch (https://github.com/Componolit/genode/commit/470d830427d141be9c63f95c7203a2af0f2ae7b7).

**Comment by jklmnn** [comment]
I could build the demo and run it on the Yoga 900.

Running the ISO on teh Yoga only spawns the terminal and a black box for the virtual box, nothing more. The terminals output is as follows:
```
[part_blk] Partition 1: LBR: 0 (22528 blocks) type: 17
[partition_report] report 'part_blk -> partitions'
[partition_report]  <partitions type="mbr">
[partition_report]    <partition number="1" type="23" start="0" length="22528"/>
[partition_report]  </partitions>
[part_blk] Error: Partition 2 unavailable for 'data_fs -> '
[data_fs] Error: Block-session creation failed (ram_quota=145408, cap_quota=5, tx_buf_size=131072, label="")
[data_fs] Error: Uncaught exception of type 'Genode::Service_denied'
[data_fs] Warning: abort called - thread: ep
```

**Comment by senier** [comment]
Great, this is the expected behavior for now. To run Android, a specific USB stick layout is required. I still need to write a README how to create such a setup (cf. senier/cbmp#104). We can test using my USB drive tomorrow.

**Comment by senier** [comment]
I encountered the missing `bin/libc_pipe.lib.so` myself. How do you think we could perform better checks - IMHO the virtualbox run files should be stricter her.

**Comment by jklmnn** [comment]
The [demo.run](https://github.com/genodelabs/genode/blob/491be000ca2c07dbda8102a6c11104230cd91ad2/repos/os/run/demo.run#L5) has a check to exit when a specific platform is set. We could run such a check if those specs are missing and return a non zero value.

**Comment by senier** [comment]
Not sure whether we can use that - I tried to check for boot/qemu being present recently (using `assert_spec`) an this didn't work. In any case, virtualbox should fail itself if we want to build it an a prerequisite is not present. This should be fixed upstream.

**Pull request closed without merging**